File: android/src/main/kotlin/com/idanalyzer/docupass/flutter/DocupassFlutterPlugin.kt
package com.idanalyzer.docupass.flutter

import io.flutter.embedding.engine.plugins.FlutterPlugin

/** Registers the DocuPass platform view factory with the Flutter engine. */
class DocupassFlutterPlugin : FlutterPlugin {
    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        binding.platformViewRegistry.registerViewFactory(
            "com.idanalyzer.docupass/view",
            DocuPassViewFactory(binding.binaryMessenger),
        )
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {}
}

File: ios/Classes/DocupassFlutterPlugin.swift
import Flutter
import UIKit

public class DocupassFlutterPlugin: NSObject, FlutterPlugin {
    public static func register(with registrar: FlutterPluginRegistrar) {
        let factory = DocuPassViewFactory(messenger: registrar.messenger())
        registrar.register(factory, withId: "com.idanalyzer.docupass/view")
    }
}
